### PR TITLE
Rtawfik/add binary reuse metal

### PR DESCRIPTION
**Pull request by rtawfik01**

### CODE REVIEW

File: tt_metal/kernels/compute/eltwise_binary.cpp [hunk 1]
```diff
@@ -18,125 +19,131 @@ void MAIN {
...
-        #endif
+#endif
 
         tile_regs_acquire();
-        for(uint32_t i = 0; i < per_core_block_size; ++i)
-        {
+
+#ifdef ELTWISE_DEST_REUSE_TYPE
+        copy_tile_to_dst_init_short();
+        for (uint32_t i = 0; i < per_core_block_size; ++i) {
```
[the remark this answers is not in the document]
Comment (rtawfik01) [reply tt_metal/kernels/compute/eltwise_binary.cpp:121]: the regular init is a "long init" that is needed once at the kernel start, the other `binary_reuse_dest_init` is a short init for configuring just the mop.

These are metal tests and I actually dont see PreScaling used in the generated binary tests

File: tests/tt_metal/tt_metal/unit_tests/compute/binary/single_core_binary_compute.cpp [hunk 2]
```diff
@@ -26,11 +26,25 @@ const map<string, string> binary_op_name_to_op_code = {
     {"add", "0"},
     {"sub", "1"},
     {"mul", "2"},
+    {"add_with_dest_reuse", "0"},
+    {"sub_with_dest_reuse", "1"},
+    {"mul_with_dest_reuse", "2"},
+};
+const map<string, string> binary_op_name_to_op_type = {
+    {"add", "EltwiseBinaryType::ELWADD"},
```
[the remark this answers is not in the document]
Comment (rtawfik01) [reply tests/tt_metal/tt_metal/unit_tests/compute/binary/single_core_binary_compute.cpp]: agreed, I cleaned that up, and also opened another issue here: https://github.com/tenstorrent/tt-metal/issues/9037
for making all the binary calls templated.

File: tt_metal/kernels/compute/eltwise_binary.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (rtawfik01) [reply tt_metal/kernels/compute/eltwise_binary.cpp]: Resolved, one of the binary_op_specific_init was not ifdef-ed out

**Review by tt-aho — APPROVED** [review]

File: tt_metal/programming_examples/eltwise_binary/eltwise_binary.cpp [hunk 3]
```diff
@@ -138,7 +138,7 @@ int main(int argc, char **argv) {
          */
         KernelHandle eltwise_binary_kernel_id = CreateKernel(
             program,
-            "tt_metal/kernels/compute/eltwise_binary.cpp",
+            "tt_eager/tt_dnn/op_library/eltwise_binary/kernels/compute/eltwise_binary.cpp",
```
Comment (tt-aho) [inline tt_metal/programming_examples/eltwise_binary/eltwise_binary.cpp:141]: Can probably leave these as the tt_metal one.

**Review by tt-aho — APPROVED** [review]

File: tt_metal/programming_examples/eltwise_binary/eltwise_binary.cpp (hunk 3 above)
Comment (rtawfik01) [reply tt_metal/programming_examples/eltwise_binary/eltwise_binary.cpp]: sorry mistake, resolved

**Review by SeanNijjar — APPROVED** [review]

**Pull request merged**